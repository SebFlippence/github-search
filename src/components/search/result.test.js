import React from 'react';
import { render } from '@testing-library/react';

import SearchResult from './result';

test('renders correctly', () => {
  const { baseElement } = render(
    <SearchResult
      id="my-id"
      login="JohnDoe"
      name="John Doe"
      email="john.doe@example.com"
      avatarUrl="https://example.com/foo.png"
    />
  );
  expect(baseElement).toMatchSnapshot();
});

test('shows user commit details are loading when clicked', async () => {
  const wrapper = render(
    <SearchResult
      id="my-id"
      login="JohnDoe"
      name="John Doe"
      email="john.doe@example.com"
      avatarUrl="https://example.com/foo.png"
    />
  );

  const button = await wrapper.findByRole('button');
  button.click();

  expect(wrapper.baseElement).toMatchSnapshot();
});
