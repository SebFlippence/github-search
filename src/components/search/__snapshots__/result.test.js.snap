// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <li
      class="py-3"
    >
      <button
        class="text-left flex items-center m-2"
        type="button"
      >
        <img
          alt="Avatar of JohnDoe"
          class="w-10 h-10 rounded-full mr-4"
          src="https://example.com/foo.png"
        />
        <div>
          <p
            class="text-gray-900 leading-none underline"
          >
            JohnDoe
          </p>
          <p
            class="text-gray-600"
          >
            John Doe
             
            <span
              class="italic"
            >
              <a
                class="hover:underline"
                href="mailto:john.doe@example.com"
              >
                john.doe@example.com
              </a>
            </span>
          </p>
        </div>
      </button>
    </li>
  </div>
</body>
`;

exports[`shows user commit details are loading when clicked 1`] = `
<body>
  <div>
    <li
      class="py-3"
    >
      <button
        class="text-left flex items-center m-2"
        type="button"
      >
        <img
          alt="Avatar of JohnDoe"
          class="w-10 h-10 rounded-full mr-4"
          src="https://example.com/foo.png"
        />
        <div>
          <p
            class="text-gray-900 leading-none underline"
          >
            JohnDoe
          </p>
          <p
            class="text-gray-600"
          >
            John Doe
             
            <span
              class="italic"
            >
              <a
                class="hover:underline"
                href="mailto:john.doe@example.com"
              >
                john.doe@example.com
              </a>
            </span>
          </p>
        </div>
      </button>
      <div
        class="border-t border-r border-b border-l border-gray-400 rounded-t rounded-b p-4"
      >
        <p
          class="italic"
        >
          Loading...
        </p>
      </div>
    </li>
  </div>
</body>
`;
